Integrations/Slack/Slack.py

```diff
@@ -0,0 +1,1095 @@
+
+
+def close_channel():
+    """
+    Archives a mirrored slack channel by its incident ID.
+    """
+    investigation = demisto.investigation()
+    if investigation.get('type') == PLAYGROUND_INVESTIGATION_TYPE:
+        return_error('Can not perform this action in playground.')
+
+    integration_context = demisto.getIntegrationContext()
+    if not integration_context or not integration_context.get('mirrors', []):
+        return_error('No mirrors found for this incident.')
+
+    mirrors = json.loads(integration_context['mirrors'])
+    mirror = list(filter(lambda m: investigation.get('id') == m['investigation_id'], mirrors))
+    if not mirror:
+        return_error('Could not find the mirrored Slack conversation.')
+
+    mirror = mirrors.pop(mirrors.index(mirror[0]))
+    conversation_id = mirror['channel_id']
+
+    CHANNEL_CLIENT.conversations_archive(channel=conversation_id)
+
+    # Check for other mirrors on the archived channel
+    channel_mirrors = list(filter(lambda m: conversation_id == m['channel_id'], mirrors))
+    for mirror in channel_mirrors:
+        mirrors.remove(mirror)
+
+    set_to_latest_integration_context('mirrors', mirrors)
+
+    demisto.results('Channel successfully archived.')
+
+
+def long_running_main():
+    """
+    Starts the long running thread.
+    """
+    asyncio.run(start_listening())
+
+
+def init_globals():
+    """
+    Initializes global variables according to the integration parameters
+    """
+    global TOKEN, CHANNEL_TOKEN, PROXY, DEDICATED_CHANNEL, CLIENT, CHANNEL_CLIENT
+    global SEVERITY_THRESHOLD, ALLOW_INCIDENTS, NOTIFY_INCIDENTS, INCIDENT_TYPE
+
+    TOKEN = demisto.params().get('bot_token')
+    CHANNEL_TOKEN = demisto.params().get('channel_token')
+    PROXY = handle_proxy().get('https')
+    DEDICATED_CHANNEL = demisto.params().get('incidentNotificationChannel')
+    CLIENT = slack.WebClient(token=TOKEN, proxy=PROXY)
+    CHANNEL_CLIENT = slack.WebClient(token=CHANNEL_TOKEN, proxy=PROXY)
+    SEVERITY_THRESHOLD = SEVERITY_DICT.get(demisto.params().get('min_severity', 'Low'), 1)
+    ALLOW_INCIDENTS = demisto.params().get('allow_incidents', False)
+    NOTIFY_INCIDENTS = demisto.params().get('notify_incidents', True)
+    INCIDENT_TYPE = demisto.params().get('incidentType')
+
+
+def main():
+    """
+    Main
+    """
+
+    init_globals()
+
+    commands = {
+        'test-module': test_module,
+        'long-running-execution': long_running_main,
+        'slack-mirror-investigation': mirror_investigation,
+        'mirror-investigation': mirror_investigation,
+        'slack-send': slack_send,
+        'send-notification': slack_send,
+        'slack-send-file': slack_send_file,
+        'slack-set-channel-topic': slack_set_channel_topic,
+        'close-channel': close_channel,
+        'slack-close-channel': close_channel
+    }
+
+    try:
+        command_func = commands[demisto.command()]
+        command_func()
+    except Exception as e:
+        return_error(str(e))
```

Comment:
We probably want to add a log of the stack trace here too. It is going to be very hard to understand errors without this.